src/aleph/http/client.clj

```diff
@@ -821,112 +821,126 @@
-
-                              (.equals ApplicationProtocolNames/HTTP_2 protocol)
-                              (do
-                                (http2/setup-conn-pipeline setup-opts)
-                                (http2-req-handler setup-opts))
-
-                              :else
-                              (do
-                                (let [msg (str "Unknown protocol: " protocol)
-                                      e (IllegalStateException. msg)]
-                                  (log/error e msg)
-                                  (netty/close ch)
-                                  (throw e))))]
-
-                    ;; Both Netty and Aleph are set up, unpause the pipeline
-                    (when (.get pipeline "pause-handler")
-                      (log/debug "Unpausing pipeline")
-                      (.remove pipeline "pause-handler"))
-
-                    (fn http-req-fn
-                      [req]
-                      (log/trace "http-req-fn fired")
-                      (log/debug "client request:" (pr-str req))
-
-                      ;; If :aleph/close is set in the req, closes the channel and
-                      ;; returns a deferred containing the result.
-                      (if (or (contains? req :aleph/close)
-                              (contains? req ::close))
-                        (-> ch (netty/close) (netty/wrap-future))
-
-                        (let [t0 (System/nanoTime)
-                              ;; I suspect the below is an error for http1
-                              ;; since the shared handler might not match.
-                              ;; Should work for HTTP2, though
-                              raw-stream? (get req :raw-stream? raw-stream?)]
-
-                          (if (or (not (.isActive ch))
-                                  (not (.isOpen ch)))
-
-                            (d/error-deferred
-                              (ex-info "Channel is inactive/closed."
-                                       {:req     req
-                                        :ch      ch
-                                        :open?   (.isOpen ch)
-                                        :active? (.isActive ch)}))
-
-                            (-> (http-req-handler req)
-                                (d/chain' (rsp-handler
-                                            {:ch                   ch
-                                             :keep-alive?          keep-alive? ; why not keep-alive?'
-                                             :raw-stream?          raw-stream?
-                                             :req                  req
-                                             :response-buffer-size response-buffer-size
-                                             :t0                   t0})))))))))))))
+               :connect-timeout     connect-timeout})
+
+        _ (attach-on-close-handler ch-d on-closed)
```

Comment:
Minor: Cleaner to use `doto` on `netty/create-client-chan` right above.

Reply:
Agreed, pushed a fixup commit to that effect!